src/undocumented_names.jl

```diff
@@ -0,0 +1,26 @@
+"""
+    test_undocumented_names(module::Module)
+
+Test that all public names in `module` have a docstring (including the module itself).
```

Comment:
Imo the module itself should be exempt bc it has an auto-generated docstring (the readme) which is often more useful than whatever short thing folks will type in to pass the aqua check

Reply:
Sounds good, I'll make the change.

Out of curiosity, how does the README go into an auto-generated docstring here? I know DocStringExtensions.jl can do that but is there another mechanism I'm not aware of?

Reply:
Ian added it to 1.7: https://github.com/JuliaLang/julia/pull/39093

Reply:
nice!